samples/net/https_client/src/main.c

```diff
@@ -304,8 +301,65 @@ int main(void)
 clean_up:
 	freeaddrinfo(res);
 	(void)close(fd);
+}
 
-	lte_lc_power_off();
+int main(void)
+{
+	int err;
+
+	printk("HTTPS client sample started\n\r");
+
+	/* Setup handler for Zephyr NET Connection Manager events. */
+	net_mgmt_init_event_callback(&l4_cb, l4_event_handler, L4_EVENT_MASK);
+	net_mgmt_add_event_callback(&l4_cb);
+
+	/* Set up IP address callback */
+	net_mgmt_init_event_callback(&ip_cb, ip_event_handler, IP_EVENT_MASK);
+	net_mgmt_add_event_callback(&ip_cb);
+
+	/* Setup handler for Zephyr NET Connection Manager Connectivity layer. */
+	net_mgmt_init_event_callback(&conn_cb, connectivity_event_handler, CONN_LAYER_EVENT_MASK);
+	net_mgmt_add_event_callback(&conn_cb);
+
+	printk("Bringing network interface up\n");
+
+	/* Connecting to the configured connectivity layer.
+	 * Wi-Fi or LTE depending on the board that the sample was built for.
+	 */
+	err = conn_mgr_all_if_up(true);
+	if (err) {
+		printk("conn_mgr_all_if_up, error: %d\n", err);
+		return err;
+	}
+
+	/* Provision certificates before connecting to the LTE network */
+	err = cert_provision();
+	if (err) {
+		return 0;
+	}
+
+	printk("Connecting to the network\n");
+
+	err = conn_mgr_all_if_connect(true);
+	if (err) {
+		printk("conn_mgr_all_if_connect, error: %d\n", err);
+		return 0;
+	}
+
+	k_sem_take(&network_connected_sem, K_FOREVER);
+
+	send_http_request();
+
+	/* A delay is in soome cases needed for the TCP stack to wrap up its work before taking down
+	 * the network interface.
+	 */
```

Comment:
`in some cases` - actually, I think it's safe to assume here, that it's reasonable to add a small delay between socket close / interface shutdown in all cases, to allow the TCP stack to gracefully teardown the connection. I'd remove that ambiguity from the comment, and just mention here that the small delay here is for the TCP connection teardown.